src/generators/default/any/index.ts

```diff
@@ -20,6 +20,15 @@ const POTENTIAL_SCHEMAS = [
 
 export const AnyGenerator = Generator({
 	schema: ZodAny,
-	output: ({ core, context }) =>
-		core.generate(core.utils.random.from(POTENTIAL_SCHEMAS), context),
+	output: ({ core, context }) => {
+		const shuffled = core.utils.random.shuffle(POTENTIAL_SCHEMAS);
+		for (const schema of shuffled) {
+			try {
+				return core.generate(schema, context);
+			} catch (e) {
+				/* empty */
+			}
+		}
+		return {};
+	},
```

Comment:
If we create a getter for the generators on core:

```ts
class Core {
    // .......
    get generators() {
        return this.#generators
    }
    // .......
}
```

then we can draw from the pool of existing generators rather than randomly trying to find a match

```ts
export const AnyGenerator = Generator({
	schema: ZodAny,
	output: ({ core, context }) => {
		const potential_schemas = new Set(core.generators);

		while (potential_schemas.size > 0) {
			const generator = core.utils.random.from(core.generators);
			potential_schemas.delete(generator);

			const schema =
				typeof generator.schema === 'function'
					? generator.schema()
					: generator.schema;

			try {
				return core.generate(schema, context);
			} catch (e) {
				/* empty */
			}
		}

		return {};
	},
});
```

The only issue here is that the `schema` is potentially incomplete if it's a "parent" type (ZodArray, ZodMap, ZodObject, etc).

```ts
z.array() // fails
z.map() // fails
z.object() // fails
```

So we'd need to add an extra method to the generator for producing a random schema.

```ts
const ArrayGenerator = Generator({
    schema: ZodArray,
    randomSchema: ({core}) => this.schema(core.utils.n(() => {
                const generator = core.utils.random.from(core.generators)
                return generator.randomSchema()
    })),
    output: () => {}
})
```

Reply:
Hrmmm... I think my goal here was to produce something that approximates an `any` value so that it would cause failures in naive implementations that didn't correctly handle the edge cases. I think I'm over-engineering here and that your implementation meets my initial requirements.

As to your latter comment, yes, we should include an `unregister` that takes the same arguments as `register` for removing a set of generators AND provide a getter for inspecting the existing generators. Let's throw that into a separate PR 😉

Reply:
Sounds good!